src/start-proxy.ts

```diff
@@ -51,10 +51,19 @@ export function getCredentials(
   }
 
   // Parse and validate the credentials
-  const parsed = JSON.parse(credentialsStr) as Credential[];
+  let parsed: Credential[];
+  try {
+    parsed = JSON.parse(credentialsStr) as Credential[];
+  } catch {
+    // Don't log the error since it might contain sensitive information.
+    logger.error("Failed to parse the credentials data.");
+    throw new Error("Invalid credentials format.");
```

Comment:
Can you convert this to a configuration error? This way, it won't count against our SLOs if this gets thrown.

```suggestion
    throw new ConfigurationError("Invalid credentials format.");
```